### PR TITLE
issue #34: allow high level config sections to be optional

### PR DESCRIPTION
**Pull request by icahoon**
These sections include:

notifications: if absent no notifcations (accounting or audit) are sent
datacenter: does not need to be present if you just want to manage dns
datacenter/compute: doesn't not need to be present if you just want to manage a network
dns: if not present, compute instances will not try to automatically create dns records
users: if users.json is not present, the don't load it, instance creation will need to deal with it

### CODE REVIEW

**Comment by icahoon** [comment]
Testing Matrix
```
test                notifications   datacenter      dns             users
-----------------   --------------  --------------  --------------  --------------
empty                 no              no              no             no
dns_only              (*)             no             yes             (*)
dc_only_nousers       (*)             yes            no              no
dc_only_users         (*)             yes            (*)             yes
dc_dns                (*)             yes            yes             (*)
all                   yes             yes            yes             yes
```

DataCenter Testing Matrix
```
test                network         compute         dns
-----------------   --------------  --------------  --------------
network_only          yes             no              (x)
network_compute       yes             yes             (x)
compute_only          no              yes             (x)             failure expected
```

File: pkg/arc/instance.go [hunk 1]
```diff
@@ -197,7 +200,11 @@ func (i *Instance) PrivateHostname() string {
 
 // PrivateFQDN returns the FQDN associated with the private ip address of the instance.
 func (i *Instance) PrivateFQDN() string {
-	return i.PrivateHostname() + "." + i.dns.Domain()
+	domain := i.Pod().Cluster().Compute().Name()
```
Comment (icahoon) [inline pkg/arc/instance.go]: This needs to be reverted. If compute is defined, dns needs to be defined.

File: pkg/arc/instance.go [hunk 2]
```diff
@@ -216,9 +223,13 @@ func (i *Instance) PublicHostname() string {
 // If this instance doesn't have a public ip address (most don't) this will return
 // an empty string "".
 func (i *Instance) PublicFQDN() string {
+	domain := i.Pod().Cluster().Compute().Name()
+	if i.Dns() != nil {
+		domain = i.Dns().Domain()
+	}
```
Comment (icahoon) [inline pkg/arc/instance.go]: This needs to be reverted. If compute is defined, dns needs to be defined.

File: pkg/arc/instance_create.go [hunk 3]
```diff
@@ -147,6 +147,14 @@ func (i *Instance) PostCreate(req *route.Request) route.Response {
 		return resp
 	}
 
+	dcName := i.Pod().Cluster().Compute().Name()
```
Comment (icahoon) [inline pkg/arc/instance_create.go]: This needs to be reverte. If compute is defined, dns needs to be defined.

File: pkg/arc/instance_users.go [hunk 4]
```diff
@@ -38,6 +38,9 @@ import (
 )
 
 func (i *Instance) configureUsers(req *route.Request, asRoot bool) route.Response {
+	if users.Groups == nil {
```
Comment (icahoon) [inline pkg/arc/instance_users.go]: Expand this to users and teams as well.

**Review by kwelch426 — APPROVED** [review]

**Comment by icahoon** [comment]
This closes issue #34

**Pull request merged**